File: buildbiblio.py
# -*- coding: utf-8 -*-
"""
Created on Wed May 29 16:04:04 2013

@author: v03nifi
"""

import os
import sys
from glob import glob

# remove all pdf-files
for f in glob ('*.pdf'):    
    os.unlink (f)

# setup typeface names and their corresponding LaTeX syntax
typeFaceName = ('computermodern','times','ebgaramond','palatino')
typeFaceSyntax = (' ','\usepackage{mathptmx}','\usepackage{ebgaramond}','\usepackage[sc]{mathpazo}')


# for all typefaces
for i in range(0,len(typeFaceName)):
    texFileName = 'biblio'
    temptexFileName = 'biblio_tmp'
    sourceFile = open(texFileName+'.tex', 'r')
    tempFile   = open(temptexFileName+'.tex', 'w')
    for line in sourceFile:
        tempFile.write(line.replace('%%stringtoberemovedbypython',typeFaceSyntax[i]))
    tempFile.close()
    sourceFile.close()
    
    # build bibliography PDF file using pdflatex and biber
    os.system('pdflatex '+ temptexFileName)
    if i ==0: # only nessecary for first compilation
        print i
        os.system('biber '+ temptexFileName)
        os.system('pdflatex '+ temptexFileName)
        os.system('pdflatex '+ temptexFileName)
    
    # rename built pdf according to its typeface
    os.rename(temptexFileName+'.pdf',texFileName+'_'+typeFaceName[i]+'.pdf')
    os.remove(temptexFileName+'.tex')


# clean up auxiliary files
os.remove(temptexFileName+'.bcf')
os.remove(temptexFileName+'.aux')
os.remove(temptexFileName+'.log')
os.remove(temptexFileName+'.out')
os.remove(temptexFileName+'.run.xml')
os.remove(temptexFileName+'.bbl')
os.remove(temptexFileName+'.blg')

# make zip file
# os.system('zip bibliography_ConcreteStructures.zip *.*')

File: buildbiblio.cpp
// c++ implementation of concrete structure's bibliography
// written by Filip Nilenius, 2014-06-03

#include <iostream>
#include <cstdlib>
#include <stdio.h>
#include <fstream>
using namespace std;


int main ()
{
    // delete old pdfs
    system("del *.pdf");

    // setup different typefaces
    string typeFaceName[] = {"computermodern","times","ebgaramond","palatino"};
    string typeFaceSyntax[] = {" ","\\usepackage{mathptmx}","\\usepackage{ebgaramond}","\\usepackage[sc]{mathpazo}"};

    // for all typefaces
    for(int n=0;n<4;++n){
        std::ifstream ifs("biblio.tex");
        std::string line;
        std::ofstream out("output.tex");
        while(std::getline(ifs, line))
        {
            if(line=="%stringtobereplacedbyfontsyntax"){
                out << typeFaceSyntax[n] <<'\n';
            }
            else{
                out << line <<'\n';
            }
        }
        out.close();

        // system calls to compile pdfs
        system("pdflatex output");
        if(n==0){ // only necessary first time
            system("biber output");
            system("pdflatex output");
        }
        system("pdflatex output");

        // rename pdf according to its typeface name
        string newName = "biblio_";
        newName.append(typeFaceName[n]);
        newName.append(".pdf");
        rename("output.pdf",newName.c_str());
        system("del output.tex");

    }
    // clean up
    system("del output.*");
    return 0;
}
